### PR TITLE
feat: smart tag support for ignoring columns on conflict

### PR DESCRIPTION
**Pull request by mgagliardo91**
This PR pulls @zebehringer 's commits from https://github.com/cdaringe/postgraphile-upsert/pull/402 that focus on enabling the feature via smart tag support:

- `@omit updateOnConflict`

The above smart tag set on columns will alter the upsert logic to skip updating these fields during an `ON CONFLICT DO UPDATE` operation. This feature is useful in restricting access to altering a column after it has already been set (during its original creation). 

Shout out to @zebehringer for putting this all together! The additional feature around input argument-driven customization can be seen in the original PR https://github.com/cdaringe/postgraphile-upsert/pull/402

### CODE REVIEW

**Comment by mgagliardo91** [comment]
@cdaringe opened this per the comment https://github.com/cdaringe/postgraphile-upsert/pull/402#issuecomment-1222409263

File: src/postgraphile-upsert.ts [hunk 1]
```diff
@@ -358,16 +363,23 @@ function createUpsertField({
...
+                (col) =>
+                  sql.query`${sql.identifier(
+                    col.names[0]
+                  )} = excluded.${sql.identifier(col.names[0])}`
+              );
 
             // SQL query for upsert mutations
             // see: http://www.postgresqltutorial.com/postgresql-upsert/
+            const conflictAction =
+              conflictUpdateArray.length == 0
```
Comment (cdaringe) [inline src/postgraphile-upsert.ts:379]:
```suggestion
              conflictUpdateArray.length === 0
```

File: src/postgraphile-upsert.ts [hunk 2]
```diff
@@ -318,8 +319,10 @@ function createUpsertField({
                 ].join(", ")}`
               );
             }
+            assert(table.namespace, "expected table namespace");
 
             const [constraintName] = matchingConstraint;
+            const ignoreUpdate = {};
```
Comment (cdaringe) [inline src/postgraphile-upsert.ts:325]: i suspect this is type `{}`, the literal type, and that `omit(...)` returns any? if so, can we type explicitly?

Comment (cdaringe) [reply src/postgraphile-upsert.ts:325]: i see what this is for. can we do something like:

```
const columnNamesSkippingUpdate = new Set<string>()
// snip snip

if (omit(...)) {
  columnNamesSkippingUpdate.add(attr.name)
}

// snip snip
.filter(col => columnNamesSkippingUpdate.has(col.names[0]))

**Review by cdaringe — APPROVED** [review]
Looks great. A few minor comments, but otherwise g2g

**Comment by mgagliardo91** [comment]
@cdaringe thanks for the review - pushed a fix with review comments addressed!

**Comment by cdaringe** [comment]
logged on to merge this, realized CI isn't running for the branch. that's on me. will probably convert this to GH actions shortly.

in the interim. the tests are failing, i believe because you need to negate `columnNamesSkippingUpdate.has(col.names[0])`

**Comment by cdaringe** [comment]
I've now setup GitHub actions. CI should run now mo' reliable :)

**Comment by mgagliardo91** [comment]
@cdaringe i don't see any status from github actions (may just be visible to you), but I fixed the expression and tests pass locally for me. Hopefully that'll be the charm!

Thanks!

**Pull request merged**